src/bin/add/args.rs

```diff
@@ -106,6 +109,22 @@ impl Args {
 
         Ok(vec![dependency])
     }
+
+    fn get_upgrade_prefix(&self) -> &'static str {
```

Comment:
If you do this, you might also want to use `self.flag_upgrade.map(|flag| match flag {…})`, which might read better than nested `match`es (and omits the redundant `None => None`).